### PR TITLE
feat: added custom name for platforms

### PR DESCRIPTION
**Pull request by zurdi15**
Now platforms can have a custom name, editing it from the platform info panel

### CODE REVIEW

File: frontend/src/components/Gallery/AppBar/Platform/PlatformInfoDrawer.vue [hunk 1]
```diff
@@ -49,13 +49,32 @@ const aspectRatioOptions = computed(() => [
...
+function toggleEditable() {
+  isEditable.value = !isEditable.value;
+}
+
+function updatePlatformName(newName: string) {
+  if (currentPlatform.value) {
+    currentPlatform.value.custom_name = newName;
+    platformApi.updatePlatform({
+      platform: {
+        ...currentPlatform.value,
```
Comment (gantoine) [inline frontend/src/components/Gallery/AppBar/Platform/PlatformInfoDrawer.vue]: you dont have to send all the fields anymore right, just the id?

File: frontend/src/components/common/Platform/Card.vue [hunk 2]
```diff
@@ -17,10 +17,10 @@ defineProps<{ platform: Platform }>();
       <v-card-text>
         <v-row class="pa-1 justify-center bg-primary">
           <div
-            :title="platform.name?.toString()"
+            :title="platform.name.toString()"
             class="px-2 text-truncate text-caption"
           >
-            <span>{{ platform.name }}</span>
+            <span>{{ platform.custom_name || platform.name }}</span>
```
Comment (gantoine) [inline frontend/src/components/common/Platform/Card.vue]: this can be done better with the pydantic resolver, i'll give it a crack tomorrow

**Review by gantoine — CHANGES_REQUESTED** [review]
can you rebase the PR to remove all the code for rom multi-select?

File: frontend/src/components/Gallery/AppBar/Platform/PlatformInfoDrawer.vue (hunk 1 above)
Comment (zurdi15) [reply frontend/src/components/Gallery/AppBar/Platform/PlatformInfoDrawer.vue]: The api method expects the platform object, so we need to send them with the updated properties

File: frontend/src/components/common/Platform/Card.vue (hunk 2 above)
Comment (zurdi15) [reply frontend/src/components/common/Platform/Card.vue]: What do you mean? we still need to send both properties to show them in the platform info panel anyway

**Review by gantoine — APPROVED** [review]

**Pull request merged**